src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,923 @@
+	Token               string   `json:"token"`               // CERN SSO token to use
+}
+
+// -------STRUCTS---------
+
+// function for get request grafana annotations endpoint for fetching annotations.
+func getAnnotations(data *[]annotationData, tags []string) error {
+
+	var headers [][]string
+	bearer := fmt.Sprintf("Bearer %s", configJSON.Token)
+	h := []string{"Authorization", bearer}
+	headers = append(headers, h)
+	h = []string{"Accept", "application/json"}
+	headers = append(headers, h)
+	// Ex: /api/annotations?from=1598366184000&to=1598366184000&tags=cmsweb&tags=cmsmon-int
+	v := url.Values{}
+
+	if trange != "" {
+		tms := parseTimes(false) // False because Listing annotations with past time offset
+
+		if len(tms) == 2 {
+			v.Add("from", strings.Trim(strconv.FormatInt(tms[0], 10), " "))
+			v.Add("to", strings.Trim(strconv.FormatInt(tms[1], 10), " "))
+		} else if len(tms) == 1 {
+			v.Add("from", strings.Trim(strconv.FormatInt(tms[0], 10), " "))
+			v.Add("to", strings.Trim(strconv.FormatInt(time.Now().UTC().Unix()*int64(milliSec), 10), " "))
+		}
+	}
+
+	for _, tag := range tags {
+		v.Add("tags", strings.Trim(tag, " "))
+	}
+
+	apiURL := fmt.Sprintf("%s%s?%s", configJSON.GrafanaBaseURL, configJSON.ListAnnotationsAPI, v.Encode())
+
+	if configJSON.Verbose > 0 {
+		log.Println(apiURL)
+	}
+	req, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		log.Printf("Unable to make request to %s, error: %s", apiURL, err)
+		return err
+	}
+	for _, v := range headers {
+		if len(v) == 2 {
+			req.Header.Add(v[0], v[1])
+		}
+	}
+
+	timeout := time.Duration(configJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+
+	if configJSON.Verbose > 1 {
+		log.Println("URL", apiURL)
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read JSON Data from Grafana Annotation GET API, error: %v\n", err)
+		return err
+	}
+
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		if configJSON.Verbose > 0 {
+			log.Println(string(byteValue))
+		}
+		log.Fatalf("Unable to parse JSON Data from Grafana Annotation GET API, error: %v\n", err)
+		return err
+	}
+
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("Response: ", string(dump))
+		}
+	}
+
+	return nil
+}
+
+// function for delete request on grafana annotations endpoint for deleting an annotation.
+func deleteAnnotationHelper(annotationID int) error {
+
+	apiurl := fmt.Sprintf("%s%s%d", configJSON.GrafanaBaseURL, configJSON.DeleteAnnotationAPI, annotationID)
+
+	req, err := http.NewRequest("DELETE", apiurl, nil)
+	if err != nil {
+		log.Printf("Request Error, error: %v\n", err)
+		return err
```

Comment:
do you really need to return the error here. Can it be replaced with log.Fatal

Reply:
fixed.